tests/src/test/scala/org/http4s/UserInfoSpec.scala

```diff
@@ -26,6 +26,14 @@ class UserInfoSpec extends Http4sSpec {
       renderString(UserInfo("hi", Some(":/?#[]@"))) must_== "hi::%2F%3F%23%5B%5D%40"
     }
 
+    "skip encoding subdelims except '+' in username" in {
+      renderString(UserInfo("!$&'()*+,;=", None)) must_== "!$&'()*%2B,;="
+    }
+
+    "skip encoding subdelims except '+' in password" in {
+      renderString(UserInfo("hi", Some("!$&'()*+,;="))) must_== "hi:!$&'()*%2B,;="
+    }
```

Comment:
These tests will break, but maybe it's good to explicitly test that all the subdelims aren't encoded.